nixos/modules/services/backup/burp.nix

```diff
@@ -0,0 +1,451 @@
+        '';
+      };
+
+      ca_name = mkOption {
+        type = types.str;
+        default = "burpCA";
+        description = ''
+          Name of CA
+        '';
+      };
+
+      ca_server_name = mkOption {
+        type = types.str;
+        default = "burpserver";
+        description = ''
+          Common name in the certificate
+        '';
+      };
+
+      ca_csr_dir = mkOption {
+        type = types.str;
+        default = "${libDir}/CA-client";
+        description = ''
+          Location of client CA certificates
+        '';
+      };
+
+      ssl_cert_ca = mkOption {
+        type = types.str;
+        default = "${libDir}/ssl_cert_ca.pem";
+        description = ''
+          Location of client SSL certificate authority -
+          should refer to the same file on server and client
+        '';
+      };
+
+      ssl_cert = mkOption {
+        type = types.str;
+        default = "${libDir}/ssl_cert-client.pem";
+        description = ''
+          Location of client SSL certificate
+        '';
+      };
+
+      ssl_key = mkOption {
+        type = types.str;
+        default = "${libDir}/ssl_cert-client.key";
+        description = ''
+          Location of client SSL key
+        '';
+      };
+
+      ssl_dhfile = mkOption {
+        type = types.str;
+        default = "${libDir}/dhfile.pem";
+        description = ''
+          Server DH file.
+        '';
+      };
+
+      ssl_key_password = mkOption {
+        type = types.str;
+        default = "change-this-password";
+        description = ''
+          SSL key password, for loading a certificate with encryption.
+        '';
+      };
+
+      ssl_peer_cn = mkOption {
+        type = types.str;
+        default = "burpserver";
+        description = ''
+          Common name in the certificate that the server gives the client
+        '';
+      };
+
+      keeps = mkOption {
+        type = types.listOf types.int;
+        default = [ 7 ];
+        description = ''
+          How many backups to keep
+        '';
+      };
+
+      timer_script = mkOption {
+        type = types.str;
+        default = "${pkgs.burp}/scripts/timer_script";
+        description = ''
+          Timer script used to evaluate if it's backup time
+        '';
+      };
+
+      timer_args = mkOption {
+        type = types.listOf types.str;
+        default = [
+          "20h"
+          "Mon,Tue,Wed,Thu,Fri,00,01,02,03,04,05,19,20,21,22,23"
+          "Sat,Sun,00,01,02,03,04,05,06,07,08,17,18,19,20,21,22,23"
+        ];
+        description = ''
+          How often should backups be triggered
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description = ''
+          Extra configuration for burp client. Contents will be added verbatim to the
+          configuration file.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.client.enable or cfg.server.enable {
```

Comment:
`or` isn't a boolean operator in Nix, check out the [list of operators](https://nixos.org/manual/nix/stable/expressions/language-operators.html). You meant to use `||` here instead. Also needs parenthesis.